src/record_stream.rs

```diff
@@ -262,9 +274,21 @@ impl RecordStream for KafkaRecordStream {
     Ok(false)
   }
 
-  async fn produce(&self, record: &[u8]) -> Result<(), RecordStreamError> {
+  async fn produce(
+    &self,
+    record: &[u8],
+    request_threshold: Option<usize>,
+  ) -> Result<(), RecordStreamError> {
     let producer = self.producer.as_ref().expect("Kafka producer not enabled");
-    let record: FutureRecord<str, [u8]> = FutureRecord::to(&self.topic).payload(record);
+    let mut record: FutureRecord<str, [u8]> = FutureRecord::to(&self.topic).payload(record);
+    if let Some(threshold) = request_threshold {
+      let threshold = (threshold as u32).to_le_bytes();
```

Comment:
I don't know anything about kafka, but it looks like this field really is just 'bytes' and it's custom serialization up to the application even for a base type like this. Is that right?

Reply:
correct, rust-rdkafka requires a `&[u8]` for header values, so we're responsible for serialization.